examples/ecs/run_conditions.rs

```diff
@@ -0,0 +1,98 @@
+//! This example demonstrates how to use run conditions to control when systems run.
+
+use bevy::prelude::*;
+
+fn main() {
+    println!();
+    println!("For the first 2 seconds you will not be able to increment the counter");
+    println!("Once that time has passed you can press space, enter, left mouse, right mouse or touch the screen to increment the counter");
+    println!();
+
+    App::new()
+        .add_plugins(DefaultPlugins.set(bevy::render::RenderPlugin {
+            wgpu_settings: bevy::render::settings::WgpuSettings {
+                backends: Some(bevy::render::settings::Backends::PRIMARY),
+                ..Default::default()
+            },
+        }))
+        .add_system(
+            increment_input_counter
+                // The common conditions module has a few useful run conditions
+                // for checking resources and states, these are included in prelude
+                .run_if(resource_exists::<InputCounter>())
+                // This is our custom run condition, both this and the
+                // above condition must be true for the system to run
+                .run_if(has_user_input),
+        )
+        .add_system(
+            print_input_counter
+                // This is also a custom run condition but this time in the form of a closure,
+                // this is useful for small, simple run conditions you don't need to reuse.
+                // All the normal rules still apply, all parameters must be read only except for local parameters
+                // In this case we will only run if the input counter resource exists and has changed but not just been added
+                .run_if(|res: Option<Res<InputCounter>>| {
```

Comment:
Am I correct in thinking that once #7605 in merged we could do something like this?
```rs
.run_if(resource_exists::<InputCounter>())
.and_then(|counter: Res<InputCounter>| counter.is_changed() && !counter.is_added())
```

Reply:
Yep, precisely :) Interestingly, the short circuiting nature would make sure that the second run condition doesn't panic. That's nice.